packages/sdk/src/actions/waitForFillTx.ts

```diff
@@ -14,11 +14,42 @@ export type WaitForFillTxParams = {
   };
   destinationChainClient: ConfiguredPublicClient;
   fromBlock: bigint;
+  depositTxHash?: Hash;
   logger?: LoggerT;
 };
 
 export async function waitForFillTx(
   params: WaitForFillTxParams,
+): Promise<FillStatus> {
+  try {
+    const statusFromFilter = await waitForFillTxEvent(params);
+    return statusFromFilter;
+  } catch (e) {
+    const message =
+      "Event filtering currently disabled for this RPC provider, switching to getFIllByDepositTx()";
+
+    params?.logger
+      ? params.logger.error(message, {
+          cause: e,
+        })
+      : console.error(message, {
```

Comment:
nit: should we instead pass in the default console logger?

Reply:
If this is being called via the client instance then the logger will be either a custom logger or the default logger and if not then we can just use the console.